### PR TITLE
chore: 🤖 git-cz integration

### PR DESCRIPTION
**Pull request by oxyii**
I like that RNE repo uses `husky`. Also I noticed that someone is trying to create semantic messages of git commits.
I would like to suggest integrating an interactive assistant [`git-cz`](https://github.com/streamich/git-cz) into RNE repo.
Initially, `git-cz` requires a global installation or `commitizen` installed globally. But after [little changes](https://github.com/streamich/git-cz/issues/79) this lib can work fine as `husky` hook executor.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -113,7 +113,8 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "lint-staged"
+      "pre-commit": "lint-staged",
+      "prepare-commit-msg": "exec < /dev/tty && node ./.git-cz/script.js || true"
```
Comment (oxyii) [inline package.json]: I'm not sure about windows users )) but linux and macOS ready

**Comment by iRoachie** [comment]
How does this work with GUI based git programs? Say like committing via VScode Git panel?

**Comment by oxyii** [comment]
@iRoachie `git-cz` is just a helper in creating a commit message. So `prepare-commit-msg` hook should return zero code anyway. That's why I wrote `... || true` in `package.json`. Without `... || true` commit failed with error `/bin/sh: /dev/tty: Device not configured` in WebStorm git GUI.

If hook returns non-zero exit-code then pre-passed commit message will pass to commit. Even in terminal, if you wrote `git commit -m "some prepared message"` but then you interrupt (`ctrl+c`) inside `git-cz` interactive shell, your commit will pass with message "some prepared message"

**Comment by iRoachie** [comment]
By looks of the config and script, this looks like something we'll have to maintain over time. Maybe we could look into [commitlint](https://github.com/conventional-changelog/commitlint?) instead.

**Comment by oxyii** [comment]
@iRoachie maintain? What exactly? Config looks simple. One-time configure `list`, `questions` and `types`... Then only manage `scopes` related to RNE components. 
`commitlint` is something else... message checker... this PR about interactive helper.
Okay, let close this PR if there are no other opinions

**Comment by iRoachie** [comment]
I'm referring to the 42k file .git-cz/script.js

**Comment by oxyii** [comment]
@iRoachie It's just build of `git-cz` with [this changes](https://github.com/streamich/git-cz/issues/79)

**Comment by iRoachie** [comment]
Really not a fan of having this in source control. Especially at that size. If you can find a npm package that abstracts having to have this file then sure.

**Comment by oxyii** [comment]
it remains to configure

**Comment by oxyii** [comment]
git-cz now [supports](https://github.com/streamich/git-cz/pull/106) husky. I will work on this PR in few days

**Pull request closed without merging**